core/quantity/src/Formatter/Format.ts

```diff
@@ -392,108 +273,15 @@ export class Format {
-      this.verifyFormatTraitsOptions(jsonObj.formatTraits); // check that all of the options for formatTraits are valid. If now, throw
-    }
-
-    if (undefined !== jsonObj.decimalSeparator) { // optional
-      if (typeof (jsonObj.decimalSeparator) !== "string") // not a string or not a one character string
-        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'decimalSeparator' attribute. It should be of type 'string'.`);
-      if (jsonObj.decimalSeparator.length !== 1)
-        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'decimalSeparator' attribute. It must be a one character string.`);
-      this._decimalSeparator = jsonObj.decimalSeparator;
-    }
-
-    if (undefined !== jsonObj.thousandSeparator) { // optional
-      if (typeof (jsonObj.thousandSeparator) !== "string")
-        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'thousandSeparator' attribute. It should be of type 'string'.`);
-      if (jsonObj.thousandSeparator.length !== 1)
-        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'thousandSeparator' attribute. It must be a one character string.`);
-      this._thousandSeparator = jsonObj.thousandSeparator;
-    }
-
-    if (undefined !== jsonObj.uomSeparator) { // optional; default is " "
-      if (typeof (jsonObj.uomSeparator) !== "string")
-        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'uomSeparator' attribute. It should be of type 'string'.`);
-      if (jsonObj.uomSeparator.length < 0 || jsonObj.uomSeparator.length > 1)
-        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'uomSeparator' attribute. It must be empty or a string with a single character.`);
-      this._uomSeparator = jsonObj.uomSeparator;
-    }
-
-    if (undefined !== jsonObj.stationSeparator) { // optional; default is "+"
-      if (typeof (jsonObj.stationSeparator) !== "string")
-        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'stationSeparator' attribute. It should be of type 'string'.`);
-      if (jsonObj.stationSeparator.length !== 1)
-        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'stationSeparator' attribute. It must be a one character string.`);
-      this._stationSeparator = jsonObj.stationSeparator;
-    }
-  }
-
   /**
    * Populates this Format with the values from the provided.
    */
   public async fromJSON(unitsProvider: UnitsProvider, jsonObj: FormatProps): Promise<void> {
     this.loadFormatProperties(jsonObj);
 
+    if (isCustomFormatProps(jsonObj))
+      this._customProps = jsonObj.custom;
```

Comment:
Do we have an example of what is put into the CustomProps?

Reply:
CustomProps is not new - it was added to support custom QuantityTypes - see [Bearing example](https://github.com/iTwin/itwinjs-core/blob/5905cb1a48c4d790b5389d7f0ea141bc3ce95f23/test-apps/ui-test-app/src/frontend/api/BearingQuantityType.ts#L13)

Reply:
One example that I've seen is the BearingFormatProps interface in the BearingQuantityType ([https://github.com/iTwin/itwinjs-core/blob/master/core/frontend/src/test/BearingQuantityType.ts]).  It's whatever extra properties that a particular formatter implementation would need to format the quantity.